Clean up ansi_c_convert_typet's interface

tautschnig opened the pull request:
The need to invoke read() was only specified in a comment. Make the
constructor take care of this. Also, make the constructor initialise all
members rather than leaving this to clear(), which is thus no longer
necessary. Finally, do not unnecessarily inherit from messaget (there is
no is-a relationship).

<!---
Thank you for your contribution. Please make sure your pull request fulfils all of the below requirements. If you cannot currently tick all the boxes, but would still like to create a PR, then add the label "work in progress" and assign the PR to yourself.
--->

- [x] Each commit message has a non-empty body, explaining why the change was made.
- n/a Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [ ] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

<!---
See, e.g., https://chris.beams.io/posts/git-commit/ for general guidelines on commit messages.

If you have created commits mixing multiple features and/or unrelated white-space changes, use a sequence involving git reset and git add -p to fix this.
--->

peterschrammel reviewed (approved):
Sounds like there should have been 3 commits, but ok.

tautschnig committed:
ansi_c_convert_typet is not a messaget

Use a message handler instead.

tautschnig committed:
Clean up ansi_c_convert_typet's interface

The need to invoke read() was only specified in a comment. Make the
constructor take care of this. Also, make the constructor initialise all
members rather than leaving this to clear(), which is thus no longer
necessary.

tautschnig commented:
> Sounds like there should have been 3 commits, but ok.

I turned it into 2 commits, breaking up the smaller one of the two turned out to be rather cumbersome and it wasn't really clear anymore where to put which change.
